README.md

```diff
@@ -0,0 +1,87 @@
+
+# Slurm charms
+
+[Juju](https://juju.is) charms for automating the Day 0 to Day 2 operations of the [Slurm workload manager](https://slurm.schedmd.com/overview.html) ⚖️🐧
+
+[![CI](https://github.com/charmed-hpc/slurm-charms/actions/workflows/ci.yaml/badge.svg)](https://github.com/charmed-hpc/slurm-charms/actions/workflows/ci.yaml/badge.svg)
+[![Release](https://github.com/charmed-hpc/slurm-charms/actions/workflows/release.yaml/badge.svg)](https://github.com/charmed-hpc/slurm-charms/actions/workflows/release.yaml/badge.svg)
+[![Matrix](https://img.shields.io/matrix/ubuntu-hpc%3Amatrix.org?logo=matrix&label=ubuntu-hpc)](https://matrix.to/#/#hpc:ubuntu.com)
+
+The `slurm-charms` repository provides many of the Slurm daemons that are required to deploy, configure and manage a Slurm-based HPC cluster:
+
+- [`slurmctld`](./charms/slurmctld/): the central management daemon for Slurm.
+- [`slurmd`](./charms/slurmd): the compute node daemon for Slurm.
+- [`slurmdbd`](./charms/slurmdbd): the database daemon for Slurm.
+- [`slurmrestd`](./charms/slurmrestd/): the REST API interface to Slurm.
+
+## ✨ Getting started
+
+This should be run using Juju 3.x or greater.
+
+```shell
+$ juju deploy slurmctld --channel edge
+$ juju deploy slurmd --channel edge
+$ juju deploy slurmdbd --channel edge
+$ juju deploy slurmrestd --channel edge
+$ juju deploy mysql --channel 8.0/stable
+$ juju deploy mysql-router slurmdbd-mysql-router --channel dpe/edge
+
+$ juju integrate slurmctld:slurmd slurmd:slurmctld
+$ juju integrate slurmctld:slurmdbd slurmdbd:slurmctld
+$ juju integrate slurmctld:slurmrestd slurmrestd:slurmctld
+$ juju integrate slurmdbd-mysql-router:backend-database mysql:database
+$ juju integrate slurmdbd:database slurmdbd-mysql-router:database
+```
+
+## 🤔 What's next?
+
+If you want to know more about the project, there are several links available for you to explore:
+
+* [Documentation](https://canonical-charmed-hpc.readthedocs-hosted.com/en/latest)
+* [Open an issue](https://github.com/charmed-hpc/slurm-charms/issues/new?title=ISSUE+TITLE&body=*Please+describe+your+issue*)
+* [Ask a question on Github](https://github.com/orgs/charmed-hpc/discussions/categories/q-a)
+
+## 🛠️ Development
+
+The project uses [tox](tox.wiki) as its command runner. It's very common to run the following four
+commands to check that the codebase is in a good state.
+
+```shell
+tox run -e fmt # Apply correct formatting to code.
+tox run -e lint # Check code against coding style standards.
+tox run -e type # Type checking.
+tox run -e unit # Run unit tests.
+```
+
+You can also run integration tests if you desire, but be aware that it requires a significant amount
+of computer resources to run them.
```

Comment:
Tbh I think we can say something along the lines of _your machine should have at least 4 cores and 16 GB of RAM to run the integration tests_. That's what we have at our disposal with the [GitHub CI runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) we're using.

This might change later down the line, but we should measure expectations with concrete numbers.

Reply:
I don't want to give specific numbers because it discourages experimentation.

Reply:
Added some numbers, but also reworded it a bit to encourage experimentation with other setups.